iothub/device/src/IotHubClientOptions.cs

```diff
@@ -69,11 +69,11 @@ public IotHubClientOptions(IotHubClientTransportSettings transportSettings)
         /// Stores custom product information that will be appended to the user agent string that is sent to IoT hub.
         /// </summary>
         public string CustomProductInfo
-        { 
+        {
             get => ProductInfo.Extra;
             set => ProductInfo.Extra = value;
         }
 
-        internal ProductInfo ProductInfo { get; } = new();
+        internal virtual ProductInfo ProductInfo { get; } = new();
```

Comment:
Q - why is this virtual?

Reply:
For mocking purposes within the SDK. While writing an MqttTransportHandler test, I needed to be able to mock this type